dhcpv6/dhcpv6message.go

```diff
@@ -229,78 +235,33 @@ func NewReplyFromRebind(rebind DHCPv6, modifiers ...Modifier) (DHCPv6, error) {
 	return d, nil
 }
 
-// NewReplyFromRenew creates a new REPLY packet based on a RENEW packet.
-func NewReplyFromRenew(renew DHCPv6, modifiers ...Modifier) (DHCPv6, error) {
-	if renew == nil {
-		return nil, errors.New("RENEW cannot be nil")
-	}
-	if renew.Type() != RENEW {
-		return nil, errors.New("The passed RENEW must have RENEW type set")
+// NewReplyFromMessageWithServerID creates a new REPLY packet based on a MESSAGE packet.
+func NewReplyFromMessageWithServerID(message DHCPv6, modifiers ...Modifier) (DHCPv6, error) {
```

Comment:
I think this function is not necessary and should be rather done by calling `NewReplyFromMessage` with a modifier, i.e.
```
reply := NewReplyFromMessage(msg, WithServerID(msg.GetOneOption(OPTION_SERVER_ID)))
```

I would leave to the user to check that the ServerID contained in msg is not `nil`